src/mongoose_rabbit_worker.erl

```diff
@@ -128,36 +133,55 @@ handle_amqp_publish(Method, Payload, Opts = #state{host = Host}) ->
     case Result of
         true ->
             update_messages_published_metrics(Host, PublishTime, Payload),
-            ?DEBUG("event=rabbit_message_sent message=~1000p", [Payload]);
+            ?DEBUG("event=rabbit_message_sent message=~1000p", [Payload]),
+            {noreply, Opts};
         false ->
             update_messages_failed_metrics(Host),
             ?WARNING_MSG("event=rabbit_message_sent_failed reason=negative_ack",
-                         []);
+                         []),
+            {noreply, Opts};
+        channel_exception ->
+            update_messages_failed_metrics(Host),
+            ?WARNING_MSG("event=rabbit_message_sent_failed reason=channel_exception",
```

Comment:
We should have description of exception returned from `publish_message_and_wait_for_confirm/3`. Here, we know that something is wrong, but do not have information about it.